src/app/student-tasks-table/student-tasks-table.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { DataSource } from '@angular/cdk/collections';
+import { MatPaginator, PageEvent } from '@angular/material';
+import { Task } from '../tasks.service';
+import { TasksService } from '../tasks.service';
+import { Observable } from 'rxjs';
+
+@Component({
+  selector: 'exc-student-tasks-table',
+  templateUrl: './student-tasks-table.component.html',
+  styleUrls: ['./student-tasks-table.component.scss']
+})
+export class StudentTasksTableComponent implements OnInit {
+  displayedColumns: string[] = ['name', 'file', 'tests', 'teacher', 'mark'];
+  dataSource: UserDataSource | null;
+  pageEvent: PageEvent;
```

Comment:
Why do you need this variable? Is it being used somewhere?

Reply:
It is used in my onPaginateChange method

Reply:
You only assign it but never read